bug: Service analyzer panics on not-ready endpoint with nil TargetRef

## Expected Behavior

A Service whose Endpoints contain a not-ready address with no `TargetRef` should be reported as a single "Service has not ready endpoints" failure, like any other not-ready endpoint.

## Current Behavior

The Service analyzer dereferences `EndpointAddress.TargetRef` while building the not-ready pod list:

```go
// pkg/analyzer/service.go
for _, addresses := range epSubset.NotReadyAddresses {
    count++
    pods = append(pods, addresses.TargetRef.Kind+"/"+addresses.TargetRef.Name)
}
```

`TargetRef` is `*v1.ObjectReference` and is marked `+optional` in `k8s.io/api`; it is nil for endpoint addresses that are not backed by a Pod (bare-IP addresses or manually-created `Endpoints`). A single such not-ready address makes the whole `k8sgpt analyze` run panic instead of reporting the Service:

```
panic: runtime error: invalid memory address or nil pointer dereference
... pkg/analyzer/service.go:99
```

## Steps to Reproduce

1. Create a Service with a selector.
2. Create an `Endpoints` object for it whose `subsets[].notReadyAddresses` contains an address with only an `ip` set (no `targetRef`), e.g. a manually-managed or bare-IP endpoint.
3. Run `k8sgpt analyze`.
4. The run panics with a nil pointer dereference at `pkg/analyzer/service.go:99`.

## Environment

- **k8sgpt version:** main (`pkg/analyzer/service.go`)
- **Kubernetes version:** n/a (reproducible against the analyzer with `k8s.io/client-go/kubernetes/fake`)
- **AI Backend/Provider:** n/a (panic happens during analysis, before any backend call)
- **OS/Platform:** any

## Additional Context

Guarding `TargetRef != nil` before appending the pod label (while still counting the address) fixes the crash and keeps the failure message unchanged. This mirrors the existing nil/empty guards added in #653 and #664.

--- a/pkg/analyzer/service.go
+++ b/pkg/analyzer/service.go
@@ -96,7 +96,9 @@ func (ServiceAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
 				if len(epSubset.NotReadyAddresses) > 0 {
 					for _, addresses := range epSubset.NotReadyAddresses {
 						count++
-						pods = append(pods, addresses.TargetRef.Kind+"/"+addresses.TargetRef.Name)
+						if addresses.TargetRef != nil {
+							pods = append(pods, addresses.TargetRef.Kind+"/"+addresses.TargetRef.Name)
+						}
 					}
 				}
 			}

--- a/pkg/analyzer/service_test.go
+++ b/pkg/analyzer/service_test.go
@@ -121,6 +121,53 @@ func TestServiceAnalyzer(t *testing.T) {
 			},
 		},
 		{
+			name: "Service with not ready endpoints with nil TargetRef",
+			config: common.Analyzer{
+				Client: &kubernetes.Client{
+					Client: fake.NewSimpleClientset(
+						&v1.Endpoints{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-service",
+								Namespace: "default",
+							},
+							Subsets: []v1.EndpointSubset{
+								{
+									// Not-ready address with no TargetRef (bare-IP /
+									// manually-created Endpoints, not backed by a Pod).
+									NotReadyAddresses: []v1.EndpointAddress{
+										{
+											IP: "10.0.0.1",
+										},
+									},
+								},
+							},
+						},
+						&v1.Service{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-service",
+								Namespace: "default",
+							},
+							Spec: v1.ServiceSpec{
+								Selector: map[string]string{
+									"app": "test",
+								},
+							},
+						},
+					),
+				},
+				Namespace: "default",
+			},
+			expectations: []struct {
+				name          string
+				failuresCount int
+			}{
+				{
+					name:          "default/test-service",
+					failuresCount: 1, // One failure for not ready endpoints
+				},
+			},
+		},
+		{
 			name: "Service with warning events",
 			config: common.Analyzer{
 				Client: &kubernetes.Client{

